### DISCUSSION TITLE
How to start?

### DISCUSSION THREAD

**Discussion by rollingafull** [Anime4000/RTL960x#33] [Q&A] [answered]
Hi,

Recently received DFP-34X-2C2 module buyed on Fibershow and i've some questions..

1º How can i look if my XPON has RTL9601D SOC or ZTE Chipset?
2º If 1º is RTL9601D, next step is flash M114_sfp_ODI_220304.tar firmware from web interface?
3º Next step is configure XPON for work with my ISP?

For more information i'm trying to configure it on Mikrotik RB5009 and CRS317.
For the moment CRS317 can't link to module, in any speed configuration and dupplex mode.

Thanks in advance!

**Comment by vuducdong** [comment]
1. you can check it with cmd in telnet

```
# cat /proc/cpuinfo
system type             : RTL8672
processor               : 0
cpu model               : 56322
BogoMIPS                : 299.00
tlb_entries             : 64
mips16 implemented      : yes
```

if it's show RTL. It's mean Realtek

**Reply by rollingafull** [reply]
Yes it is RTL, thanks.

Now can i flash latest firmware across web interface?

regards.

**Reply by vuducdong** [reply]
> Yes it is RTL, thanks.
> 
> Now can i flash latest firmware across web interface?
> 
> regards.

Yes, of course

Some routers need manual setup (turn off auto negotiation) and set 1G or 2.5G full duplex

```
#Verify software version
omcicli mib get 7

#Verify ONT attributes
omcicli mib get 257

you can see RTL9601D is here
```

**Reply by rollingafull** [reply]
Flashed!

Thank you so much! 

After your commands,

> #Verify software version
> omcicli mib get 7
> 
> #Verify ONT attributes
> omcicli mib get 257
> 
> you can see RTL9601D is here

RTL9601D is not showed.

`# omcicli mib get 7
XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
SWImage
XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
=================================
EntityID: 0x00
Active: 0
Committed: 0
Valid: 1
Version: V1.0-220304
ProductCode:
ImageHash: 0x00000000000000000000000000000000
=================================
=================================
EntityID: 0x01
Active: 1
Committed: 1
Valid: 1
Version: V1.0-220304
ProductCode:
ImageHash: 0x00000000000000000000000000000000
=================================
# omcicli mib get 257
XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
Ont2g
XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
=================================
EntityID: 0x0000
EqtID: 0x4446502d3334582d324332000000000000000000
OMCCVer: 0x80
VPCode: 15
SecCapability: 1
SecMode: 1
NumOfPriQ: 128
NumOfScheduler: 16
Mode: 1
NumOfGemPort: 64
SysUpTime: 1500
ConnectivityCapability: 0x007f
CurrentConnectivityMode: 0
QosConfigurationFlexibility: 0x003b
PriorityQueueScaleFactor: 1
=================================
`

i'm doing something wrong?

regards.

**Comment by tdmadam** [comment]
Don't worry about it. If it is Realtek than the SoC is RTL9601D
As a next step, I recommend backing up your configuration. Preferably mtd1/mtd2/mtd3. At least store the output text of these two commands somewhere safe. It contains the original MAC_KEY, MAC address and other important variables.

```
# nv getenv
# flash all
```

**Reply by rollingafull** [reply]
Hi,

Thanks, i'll do it.

For now i updated ROS to 7.3beta40 and lost conexión with ODI Module :-(

Somebody with same issue?

**Comment by rollingafull** [comment]
Hi,

I don't know wha's the matter.

I has 3 devices:
RB5009 link ok with ODI module 1Gbs.
CRS317 no link with ODI module.
CRS326-24g-2s+ no link with ODI module.

I has configured spf-plus port into bridge lan and put mi pc in same ip range 192.168.1.x/24, 1 week ago i acces to ODI and flash latest firmware ok, it restart an i access to module with new firmware.

ok, i left it disconected for the weekend beacouse i stay out of home.

From tuesday doins the same process and config on Mikrotik devices i cant access to ODI module, i not answer to ping (192.168.1.1) and not load web config page.

Please, any idea that can i do to access to the module?

Why it not link on CRS devices? specialy on CRS326 tha has same cpu than CRS305 that works well.

I try with ROS 7.2.3 and 7.3beta....

Thanks in advance.

**Reply by Anime4000** [reply]
I afraid both not support `1000baseX` or `SGMII`
if RB5009 works, try check which LAN SDS in your RB5009

Here ODI stick in my RB5009
```
# cat /proc/lan_sds/lan_sds_cfg
lan_sds_mode = 3(SGMII MAC)
# flash get LAN_SDS_MODE
LAN_SDS_MODE=0
```

you might try set `flash set LAN_SDS_MODE`, refer [documentation here](https://github.com/Anime4000/RTL960x/blob/main/Docs/FLASH_GETSET_INFO.md#lan_sds_mode-min-1-max-6):

**Reply by rollingafull** [reply]
Hi,

thanks.

First i reboot the rb5009 and set flow control from auto --> off and recover the connection with XPOM module.

This is my config.
`# cat /proc/lan_sds/lan_sds_cfg`
`lan_sds_mode = 7(SGMII Force)`
`# flash get LAN_SDS_MODE`
`LAN_SDS_MODE=3`

@Anime4000 in your experience, do you think that is posible `set LAN_SDS_MODE=5` for work at 2'5Gb on RB5009? Maybe in ROS 7.3beta?
For work with CRS3xx i think the config must be `LAN_SDS_MODE=4`. Is this correct?

If i put `LAN_SDS_MODE=5` and RB5009 not support for now, i will lost connection until Mikrotik solve it not?

i've fear LOL.

Thanks for all help community.

**Reply by Anime4000** [reply]
My RB5009 was in `off` Auto Flow

2.5Gb is not supported in RB5009, it has missing `HSGMII` and `2500baseX`,
if you try, high chance you cant recover it,

if dont have 2.5Gb SFP supported host, you need USB-UART ODI to revert back settings

based on [@stich86](https://github.com/stich86), he use `HiSGMII PHY` mode (`4`)
you can try `LAN_SDS_MODE 2` first

**Comment by tdmadam** [comment]
One other thing I wanted to mention is the different behavior of my ODI stick when it comes to responding to pings and connectivity in general. It's like it has some kind of power saving option implemented. I don't have this problem with V2801F

The tests were conducted with the fiber optic cable disconnected.

```
0sec - Power on
8sec - ODI module link is UP does not respond to pings
58sec - ODI module link DOWN
1m03sec - ODI module link UP ping 192.168.1.1 responds
1m20sec - ODI module link DOWN
2m25sec - ODI module link UP
2m40sec - ODI module link DOWN
```

I don't know how it behaves in Mikrotik but in my example if I don't bring the Ethernet interface on my computer UP during the 15 seconds window, I could not connect to it.

**Reply by Anime4000** [reply]
it seem ODI auto detect host have a problem,

try this
```
flash set LAN_SDS_MODE 1
reboot
```

This make your ODI follow V2801F mode (1000baseX)

**Comment by tdmadam** [comment]
Mine was LAN_SDS_MODE = 0

After changing to 1 I was able to ping after 35 seconds, but there still seems to be some watchdog. After 1m20sec without turning the interface UP, it went DOWN, only to be available again at 2m25sec for a short period of time.

But I don't use auto-config because I have several different profiles on my Ubuntu 192.168.1.x, 192.168.88.x, 192.168.2.x, 10.10.10.x and I need to choose the correct one.
For users who only have 192.168.1.x then maybe this is not a problem.

**Comment by rollingafull** [comment]
Hi,

DFP-34X-2C2 must be configured on router mode?

![imagen](https://user-images.githubusercontent.com/62690523/169800481-b986021c-0804-4ff5-8fcf-b0bcfcc3240c.png)

This parameters must be configured like red arrow shows?

Another issue is that after set OMCI_SW_VER1 and OMCI_SW_VER2 and reboot it not will setted pemanently.

Thanks.

**Reply by Anime4000** [reply]
older ODI Firmware was in Router Mode
ODI Firmware 220304 and above now in Bridge Mode just like V2801F, TWCGPON657.

changing software version in ODI is know issues revert back to original value, probably hardcoded inside binary program cause it,

you can try use OMCI_OLT_MODE 1 or 2 or 3 if software version can be change,

work around is you need a Linux Desktop like Ubuntu and modify the firmware

**Comment by rollingafull** [comment, marked as answer]
Hi!

now i've link with with O5, but stay in initial status.
![PON_Status](https://user-images.githubusercontent.com/62690523/171352337-dce24680-9c85-47f0-b232-66f13e59321a.png)

With this status i can't put up the PPPoE across vlan.

For more info, my ISP install me one ONT Nokia G-010G-PL.
I try whith a Huawei HG8240H, and it works ok only putting PLOAM_PASSWD.

My ODI stick shows same status configured like Nokia config and Huawei config.

Any idea about what can i try?

Thanks.

**Reply by vuducdong** [reply]
> Hi!
> 
> now i've link with with O5, but stay in initial status. ![PON_Status](https://user-images.githubusercontent.com/62690523/171352337-dce24680-9c85-47f0-b232-66f13e59321a.png)
> 
> With this status i can't put up the PPPoE across vlan.
> 
> For more info, my ISP install me one ONT Nokia G-010G-PL. I try whith a Huawei HG8240H, and it works ok only putting PLOAM_PASSWD.
> 
> My ODI stick shows same status configured like Nokia config and Huawei config.
> 
> Any idea about what can i try?
> 
> Thanks.

about OLT Alcatel/Nokia, when you use SFP GPON, no matter you fill it right, or fill it wrong, or fill in any parameter, the state is always O5 (GPON SN, PLOAM_PASSWD etc...)

It seems that now will need extra parameters such as: OMCC version, OMCI version, Hardware version, Vendor ID ... only when you are correct will you be able to see the vlan (there will be 2 cases you see: original vlan from OLT or converted vlan on ONT side-if yes, like when you use your ISP's original ONT).

for this case, usually I will use nokia sfp like g-010s-a then it should be ok.

**Reply by Anime4000** [reply]
Your ONU Status `O5` is same like my ISP OLT, even with default ODI value give me `O5`.

after reading @vuducdong about Alcatel/Nokia...

Now I can confirmed Alcatel/Nokia OLT giving fake ONU Status, plus ISP technician told me I connected to Nokia OLT
Gladly my Nokia OLT not strict as yours, I just need PLOAM Password and `flash set HW_HWVER 168D.A`:
![image](https://user-images.githubusercontent.com/1908715/171401009-d253908f-931c-4a1d-9a80-75ce6ad6d9f4.png)

You can try `OMCC version` first, once connected, do a speedtest then try change `OMCI_TM_OPT` to get highest speed

**Reply by rollingafull** [reply]
Hi guys!!!

I'm connected across ODI DFP-34X-2C2 with my ISP 'Lowi' in Spain.
The solution over the previous question will change the vlan id.
With the Nokia and Huawei ONU i must use VLAN ID 24, but with DFP-34X-2C2 i must use VLAN ID 179 like shows next code.

Only change this in my Mikrotik RB5009 and voilá.

`# omcicli mib get 171`
`XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX`
`ExtVlanTagOperCfgData`
`XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX`
`# omcicli mib get 84`
`XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX`
`VlanTagFilterData`
`XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX`
`=================================`
`EntityID: 0x01`
`FilterTbl[0]: PRI 0,CFI 0, VID 179`
`FwdOp:  0x04`
`NumOfEntries: 1`
`=================================`
`#`

@Anime4000 if you want any info for do any page with a list of configs params for ISPs arround the world, only tell me.

But i've one more question. How you can see this info in your module and i can't?

![imagen](https://user-images.githubusercontent.com/62690523/171471013-39f876cd-71f5-463e-a079-0b23ee4a7b2a.png)

![imagen](https://user-images.githubusercontent.com/62690523/171471663-6e16c3b2-9167-436e-a6aa-f451afa673d3.png)

Thank you so much for all your help!

![imagen](https://user-images.githubusercontent.com/62690523/171471242-47db5b97-95ae-4fa3-8219-42a490b0c3ef.png)

**Reply by Anime4000** [reply]
Nice! you get it connected

Lantiq Stick can show SFP Properties
Realtek Stick kind broken, you need login into WebGUI to view PON signal
